src/cargo/core/dependency.rs

```diff
@@ -103,6 +105,25 @@ impl DependencyInner {
         })
     }
 
+    fn parse_with_deprecated(req: &str, config: &Config) -> CargoResult<VersionReq> {
+        match VersionReq::parse(req) {
+            Err(e) => {
+                match e {
+                    ReqParseError::DeprecatedVersionRequirement(requirement) => {
+                        let msg = format!("One of your version requirements ({}) is invalid. \
+Previous versions of Cargo accepted this malformed requirement, but it is being deprecated. Please \
+use {} instead.", req, requirement);
```

Comment:
Ah yeah I'm just worried about users looking at this error and not really knowing what to do when the error shows up. The current versions says "one of you version requirements" but that's actually not true if it's coming from some upstream dependency (and then you have to hunt and track that down)